sourcemap/store_test.go

```diff
@@ -142,6 +153,145 @@ func TestStore_Fetch(t *testing.T) {
 	})
 }
 
+func TestFetchTimeout(t *testing.T) {
+	var (
+		errs int64
+
+		apikey  = "supersecret"
+		name    = "webapp"
+		version = "1.0.0"
+		path    = "/my/path/to/bundle.js.map"
+		c       = http.DefaultClient
+		waitc   = make(chan struct{})
+	)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		<-waitc
+		// zlib compress
+		wr := zlib.NewWriter(w)
+		defer wr.Close()
+		wr.Write([]byte(test.ValidSourcemap))
+	}))
+	defer ts.Close()
+
+	fleetCfg := &config.Fleet{
+		Hosts:        []string{ts.URL},
+		Protocol:     "https",
+		AccessAPIKey: apikey,
+		TLS:          nil,
+	}
+	cfgs := []config.SourceMapMetadata{
+		{
+			ServiceName:    name,
+			ServiceVersion: version,
+			BundleFilepath: path,
+			SourceMapURL:   "",
+		},
+	}
+	b, err := newFleetStore(c, fleetCfg, cfgs)
+	assert.NoError(t, err)
+	logger := logp.NewLogger(logs.Sourcemap)
+	store, err := newStore(b, logger, time.Minute)
+	assert.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+	defer cancel()
+
+	var wg sync.WaitGroup
+	for i := 0; i < 2; i++ {
+		wg.Add(1)
+		go func() {
+			consumer, err := store.Fetch(ctx, name, version, path)
+			if err != nil {
+				assert.True(t, errors.Is(err, context.DeadlineExceeded))
+				atomic.AddInt64(&errs, 1)
+				close(waitc)
+			} else {
+				assert.NotNil(t, consumer)
+			}
+
+			wg.Done()
+		}()
+	}
+
+	wg.Wait()
+	assert.Equal(t, int64(1), errs)
```

Comment:
I feel like this test could be simplified a bit. Are multiple goroutines necessary? Can't we just make a single Fetch call in the main test function?